plugins/scorecards/src/api/ScorecardsClient.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ScorecardsApi } from './ScorecardsApi';
+import { CheckResult } from '@backstage/plugin-tech-insights-common';
+import { Check } from './types';
+import { DiscoveryApi } from '@backstage/core-plugin-api';
+import {
+  CheckResultRenderer,
+  defaultCheckResultRenderers,
+} from '../components/CheckResultRenderer';
+
+export type Options = {
+  discoveryApi: DiscoveryApi;
+  proxyPath?: string;
+};
+
+export class ScorecardsClient implements ScorecardsApi {
+  private readonly discoveryApi: DiscoveryApi;
+  private baseUrl: string = '';
+
+  constructor(options: Options) {
+    this.discoveryApi = options.discoveryApi;
+  }
+
+  getScorecardsDefinition(
+    type: string,
+    value: CheckResult[],
+  ): CheckResultRenderer | undefined {
+    const resultRenderers = defaultCheckResultRenderers(value);
+    return resultRenderers.find(d => d.type === type);
+  }
+
+  getBaseUrl: () => Promise<string> = async () => {
+    if (!this.baseUrl) {
+      this.baseUrl = await this.discoveryApi.getBaseUrl('tech-insights');
```

Comment:
The discovery has as part of its contract that it should be possible to call it repeatedly without worry - the default one indeed only performs a string concatenation and returns that. You should not need to cache this output, and doing so could even lead to errors, since a more competent discovery API may return dynamic results that can change as your target deployments change at runtime and similar.